libs/payments/stripe/src/lib/stripe.manager.ts

```diff
@@ -146,4 +153,42 @@ export class StripeManager {
 
     return taxIdFields.at(0)?.value;
   }
+
+  async getPlan(planId: string) {
+    const plan = await this.client.plansRetrieve(planId);
+    if (!plan) throw new PlanNotFoundError();
+    return plan;
+  }
+
+  async getPlanByInterval(planIds: string[], interval: string) {
+    const plans = [];
+    for (const planId of planIds) {
+      const plan = await this.getPlan(planId);
+      if (plan.interval === interval) {
+        plans.push(plan);
+      }
+    }
+    if (plans.length > 1) throw new PlanIntervalMultiplePlansError();
+    return plans.at(0);
+  }
+
+  /**
+   * Returns array of customer subscription plans
+   */
+  getSubscribedPlans(subscriptionList: StripeApiList<StripeSubscription>) {
+    const subscriptions = subscriptionList.data;
+    if (!subscriptions) return [];
+    return subscriptions
+      .flatMap((sub) => sub.items.data)
+      .map((item) => item.plan as StripePlan);
+  }
+
+  /**
+   * Returns array of subscribed product IDs
+   */
+  getSubscribedProductIds(subscribedPlans: StripePlan[]) {
+    return subscribedPlans.length > 0
+      ? subscribedPlans.map(({ product: productId }) => productId as string)
+      : [];
+  }
```

Comment:
These two methods don't depend on anything in the class and can be moved out into a `util` directory with each of them in their own file.

Reply:
Sorry @xlisachan , could you move them back to utils functions? I spoke with Ben about this as well, and agreed it makes more sense as a util.

Reply:
No worries, I'll move it back! Thanks!